File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        curr = 0
        ans = 0
        for i in nums:
            if i == 0:
                curr += 1
            else:
                ans += ((curr+1)*curr//2)
                curr = 0
        ans += ((curr+1)*curr//2)
        return ans

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.rs
impl Solution {
    pub fn zero_filled_subarray(nums: Vec<i32>) -> i64 {
        let mut curr = 0;
        let mut ans = 0;
        for i in nums{
            if i == 0{
                curr += 1;
            }
            else{
                ans += ((curr+1)*curr/2);
                curr = 0;
            }

            }
        ans + ((curr+1)*curr/2)
    }
}